parsing/parser.mly

```diff
@@ -458,6 +458,7 @@ let extra_rhs_core_type ct ~pos =
 type let_binding =
   { lb_pattern: pattern;
     lb_expression: expression;
+    lb_is_pun: bool;
```

Comment:
I we are going to have this wart in the codebase, let's turn it into a strength that will make ocamlformat people happy: use it also for binding operators, and use this boolean when reprinting the AST.

Reply:
I'm not quite sure what you're suggesting here? The type `let_binding` is not part of the AST, but rather a temporary structure used by `parser.mly` to share code between various constructs that have similar syntax.